addon/index.js

```diff
@@ -58,8 +47,17 @@ export {
   getPretender,
 };
 
+/**
+ * Setup and teardown code, intended to be called with qunit hooks so that it can run code before & after each test.
+ */
 export function setupFactoryGuy(hooks) {
   hooks.beforeEach(function () {
-    manualSetup(this);
+    const owner =
+      this.owner || (this.container && this.container.owner) || this;
```

Comment:
would have liked to use `getOwner(this)` because simpler, but it doesn't seem available (guessing because ember 3.28? I think it was introduced in 4.x?)

If we end up dropping ember 3.x I'll look into replacing this